net/net.go

```diff
@@ -276,6 +276,10 @@ func (n *net) AddThread(ctx context.Context, addr ma.Multiaddr, opts ...core.New
 		}
 	}
 
+	if !id.Valid() {
```

Comment:
Should this be checked at the top of these public API methods?

Reply:
Yea good call. So in this file, any method that accepts or derives a thread id from data coming from the api user, I'll call `Validate`